lib/rules/jsx-no-target-blank.js

```diff
@@ -88,6 +110,17 @@ function hasSecureRel(node, allowReferrer, warnOnSpreadAttributes, spreadAttribu
 
   const relAttribute = node.attributes[relIndex];
   const value = getStringFromValue(relAttribute.value);
+  if (Array.isArray(value)) {
+    for (let index = 0; index < value.length; index++) {
+      const tags = value[index] && typeof value[index] === 'string' && value[index].toLowerCase().split(' ');
+      const noreferrer = tags && tags.indexOf('noreferrer') >= 0;
+      const noopener = tags && tags.indexOf('noopener') >= 0;
+      if (!noreferrer) {
+        return allowReferrer && noopener;
+      }
+    }
+    return true;
```

Comment:
```suggestion
    return [].concat(value).filter(Boolean).some((item) => {
      const tags = typeof item === 'string' && item.toLowerCase().split(' ');
      const noreferrer = tags && tags.indexOf('noreferrer') >= 0;
      const noopener = tags && tags.indexOf('noopener') >= 0;
      if (!noreferrer) {
        return allowReferrer && noopener;
      }
      return false;
    });
```
also the [].concat means you don't have to check if it's an array; you can just use this code for both the single and the multi case.

Reply:
instead of `some(item)`, I think having `every(item)` would be workable since every value should be true; But I don't why the test cases are failing because of that. If `some(item)` is kept, then as soon it would get first true value it won't check the rest and there would give no errors for the cases like `rel={isExternal ? "noreferrer" : "3"}`

Reply:
I agree with you that every is the proper choice